File: WebAPI/Exam_sch-system-WebApi/Exam_sch-system-WebApi/Models/RoomDetail.cs
using System;
using System.Collections.Generic;

namespace Exam_sch_system_WebApi.Models;

public partial class RoomDetail
{
    public int RoomDetailsId { get; set; }

    public string ColumnName { get; set; } = null!;

    public int RowCapacity { get; set; }

    public int? RoomId { get; set; }

    public virtual ICollection<Room> Rooms { get; } = new List<Room>();
}

File: src/app/Models/RoomDetail.ts
export interface Room {
  roomId: number;
  roomName: string;
  seatNumber: number;
  columns: number;
  building: string;
  roomDetailsId?: number | null;
  roomDetails?: RoomDetail | null;
}

export interface RoomDetail {
  roomDetailsId: number;
  columnName: string;
  rowCapacity: number;
  roomId: number;
}
export interface RoomDetails {
    columnName: string;
    rowCapacity: number;
}
